pkg/client/container_repository_client.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/oslokommune/okctl/pkg/api"
+	"github.com/oslokommune/okctl/pkg/client/store"
+)
+
+// CreateContainerRepositoryOpts contains necessary information to create a container repository
+type CreateContainerRepositoryOpts struct {
+	ClusterID api.ID
+	ImageName string
+}
+
+// DeleteContainerRepositoryOpts contains necessary information to delete a container repository
+type DeleteContainerRepositoryOpts struct {
+	ClusterID api.ID
+	ImageName string
+}
+
+// ContainerRepository contains state after creating a container repository
+type ContainerRepository struct {
+	ClusterID              api.ID
+	ImageName              string
+	StackName              string
+	CloudFormationTemplate string
+}
+
+// ContainerRepositoryService orchestrates the creation of various services
+type ContainerRepositoryService interface {
+	CreateContainerRepository(ctx context.Context, opts CreateContainerRepositoryOpts) (*ContainerRepository, error)
+	DeleteContainerRepository(ctx context.Context, opts DeleteContainerRepositoryOpts) error
+}
+
+// ContainerRepositoryAPI invokes the API
+type ContainerRepositoryAPI interface {
+	CreateContainerRepository(opts api.CreateContainerRepositoryOpts) (*api.ContainerRepository, error)
+	DeleteContainerRepository(opts api.DeleteContainerRepositoryOpts) error
+}
+
+// ContainerRepositoryStore saves the data
+type ContainerRepositoryStore interface {
+	SaveContainerRepository(repository *ContainerRepository) (*store.Report, error)
+	RemoveContainerRepository(imageName string) (*store.Report, error)
+}
+
+// ContainerRepositoryState updates the state
+type ContainerRepositoryState interface {
+	SaveContainerRepository(repository *ContainerRepository) (*store.Report, error)
+	RemoveContainerRepository(imageName string) (*store.Report, error)
+	GetContainerRepository(imageName string) (*ContainerRepository, error)
+}
+
+// ContainerRepositoryReport reports on the state and storage operations
+type ContainerRepositoryReport interface {
```

Comment:
You can keep it if you want to, but I will remove it in my state PR, so if you waaaant to 😬